cargo-shuttle/src/lib.rs

```diff
@@ -337,42 +337,53 @@ impl Shuttle {
                 It will be hosted at ${{project_name}}.shuttleapp.rs, so choose something unique!
                 "
             };
-            let client = self.client.as_ref().unwrap();
-            loop {
-                // not using validate_with due to being blocking
-                let p: String = Input::with_theme(&theme)
+        }
+        let mut prev_name: Option<String> = None;
+        loop {
+            // prompt if interactive
+            let name: String = if let Some(name) = project_args.name.clone() {
+                name
+            } else {
+                // not using `validate_with` due to being blocking. we also need to handle --force
+                Input::with_theme(&theme)
                     .with_prompt("Project name")
-                    .interact()?;
-                match client.check_project_name(&p).await {
-                    Ok(true) => {
-                        println!("{} {}", "Project name already taken:".red(), p);
-                        println!("{}", "Try a different name.".yellow());
-                    }
-                    Ok(false) => {
-                        project_args.name = Some(p);
-                        break;
-                    }
-                    Err(e) => {
-                        // If API error contains message regarding format of error name, print that error and prompt again
-                        if let Ok(api_error) = e.downcast::<ApiError>() {
-                            // If the returned error string changes, this could break
-                            if api_error.message.contains("Invalid project name") {
-                                println!("{}", api_error.message.yellow());
-                                println!("{}", "Try a different name.".yellow());
-                                continue;
-                            }
-                        }
-                        // Else, the API error was about something else.
-                        // Ignore and keep going to not prevent the flow of the init command.
-                        project_args.name = Some(p);
-                        println!(
-                            "{}",
-                            "Failed to check if project name is available.".yellow()
-                        );
-                        break;
-                    }
+                    .interact()?
+            };
+            // handle --force on retires
+            let name: String = if let Some(prev_name) = prev_name {
+                if project_args.name.is_none() && name == "--force" {
```

Comment:
I'm lost. :confused: Where does `--force` come from? The user's input?

Reply:
Yes, see the prompt that gets printed if the check fails.

Reply:
Hmm, that is not good DX imo. The `force-name` argument should be enough and the error can suggest users use it instead.

Reply:
Disagree. Forcing them to quit the command and add arguments instead of resolving it in interactive mode is worse.

Reply:
Else there should be a control key or force prompt instead. Having `--force` as a name input is just not going to cut it

Reply:
A separate prompt hinders the normal flow of letting user write a different project name instead. The --force is the more rare use case.

Reply:
Then maybe a control key like "Ctrl+f" to use it anyway

Reply:
Or ask the user to type the same name if they want to force use it

Reply:
I like it.